packages/core/src/services/essentials/dom-utils.ts

```diff
@@ -0,0 +1,205 @@
+				width: Tools.getProperty(svgSelector.node(), "width", "baseVal", "value"),
+				height: Tools.getProperty(svgSelector.node(), "height", "baseVal", "value")
+			};
+			if (validateDimensions(nativeDimensions)) {
+				return nativeDimensions;
+			}
+		} catch (e) {
+			if (validateDimensions(clientDimensions)) {
+				return clientDimensions;
+			}
+
+			if (validateDimensions(bboxDimensions)) {
+				return bboxDimensions;
+			}
+
+			if (validateDimensions(attrDimensions)) {
+				return attrDimensions;
+			}
+		}
+
+		return {
+			width: 0,
+			height: 0
+		};
+	}
+
+	static appendOrSelect(parent, query) {
+		const querySections = query.split(".");
+		const elementToAppend = querySections[0];
+
+		const selection = parent.select(query);
+		if (selection.empty()) {
+			return parent.append(elementToAppend)
+				.attr("class", querySections.slice(1).join(" "));
+		}
+
+		return selection;
+	}
+
+	protected svg: Element;
+
+	init() {
+		// Add width & height to the chart holder if necessary, and add a classname
+		this.styleHolderElement();
+
+		// Add main SVG
+		this.addSVGElement();
+
+		if (this.model.getOptions().resizable) {
+			this.addResizeListener();
+		}
+	}
+
+	styleHolderElement() {
+		const holderElement = this.model.get(ModelStateKeys.HOLDER) as HTMLElement;
+		const { width, height } = this.model.getOptions();
+
+		// Add class to chart holder
+		select(this.model.get(ModelStateKeys.HOLDER)).classed(`${settings.prefix}--chart-holder`, true);
+
+		// If width exists in options
+		if (width) {
+			// Apply formatted width attribute to chart
+			holderElement.style.width = Tools.formatWidthHeightValues(width);
+		}
+
+		// If height exists in options
+		if (height) {
+			// Apply formatted height attribute to chart
+			holderElement.style.height = Tools.formatWidthHeightValues(height);
+		}
+	}
+
+	getHolder() {
+		return this.model.get(ModelStateKeys.HOLDER);
+	}
+
+	addSVGElement() {
+		const chartsprefix = Tools.getProperty(this.model.getOptions(), "style", "prefix");
+		const svg = select(this.model.get(ModelStateKeys.HOLDER))
+			.append("svg")
+			.classed(`${settings.prefix}--${chartsprefix}--chart-svg`, true)
+			.attr("height", "100%")
+			.attr("width", "100%");
+
+		this.svg = svg.node();
+	}
+
+	getMainSVG() {
+		return this.svg;
+	}
+
+	addResizeListener() {
+		const holder = this.model.get(ModelStateKeys.HOLDER);
+
+		if (!holder) {
+			return;
+		}
+
+		// Grab current dimensions of the chart holder
+		let containerWidth = holder.clientWidth;
+		let containerHeight = holder.clientHeight;
+
+		// The resize callback function
+		const resizeCallback = Tools.debounce((entries, observer) => {
+			if (!holder) {
+				return;
+			}
+
+			if (Math.abs(containerWidth - holder.clientWidth) > 1
+				|| Math.abs(containerHeight - holder.clientHeight) > 1) {
+				containerWidth = holder.clientWidth;
+				containerHeight = holder.clientHeight;
+
+				this.services.events.dispatchEvent("chart-resize");
+			}
+		}, 12.5);
```

Comment:
Why `12.5`?

Reply:
You can add any value in there. `12.5` produces smooth resizes in my trials while still not calling the resize function too many times